### ISSUE TITLE
Unable to authenticate via kubernetes after upgrading bank-vaults 0.4.6 -> 0.5.1 and vault 1.0.2 -> 1.2.2

### ISSUE THREAD

**Issue by jordan-next** [banzaicloud/bank-vaults#657]
We had Vault Operator Helm Chart v0.2.6, with Bank-Vaults v0.4.6 and Vault v1.0.2 installed on our kubernetes cluster.  We wanted to upgrade vault to a later version and upgrade everything else along with it.  I ran the upgrade locally to make sure it could go smoothly and then upgraded out cluster.

I first ran `kubectl apply` with my CR to use Bank-Vaults v0.5.1 and Vault v1.2.2, and then ran `helm upgrade --install --namespace kube-system --version 0.4.1 vault-operator banzaicloud-stable/vault-operator` to upgrade the helm chart.

We had to regenerate a root token after the upgrade, but that maybe have been because I didn’t shut down vault before the upgrade.

Beyond that issue it seemed to go smoothly, I was able to login with a root token and all my secrets are still there, however we had issues with our kubernetes login.

**Describe the bug**:
Trying to login via kubernetes gives this error
```{
    "errors": [
        "{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"tokenreviews.authentication.k8s.io is forbidden: User \\\"system:serviceaccount:kube-system:vault-operator\\\" cannot create tokenreviews.authentication.k8s.io at the cluster scope\",\"reason\":\"Forbidden\",\"details\":{\"group\":\"authentication.k8s.io\",\"kind\":\"tokenreviews\"},\"code\":403}"
    ]
}
```

When trying to connect to vault via kubernetes and Spring Boot we get this error

`org.springframework.web.client.ResourceAccessException: I/O error on POST request for “{vault-path}/v1/auth/kubernetes/login": null; nested exception is org.apache.http.client.ClientProtocolException` which was working before with no code changes.

Inside the vault-configurer pod we saw this

We believe our issue has to do with missing cluster roles definitions. Compared to our other environments one big thing missing is vault-operator-auth-delegator which was removed in [this commit](https://github.com/banzaicloud/bank-vaults/commit/c54a44c4bd218e627f86e9b4d55091d4e0608f84#diff-4a14cb2f882819d989ec3d03a66db72b).  Is this something that is no longer needed or needs to be created in the newer vault CRs?

We were able to resolve our issue by creating a clusterrolebinding for the vault-operator, but we are not sure why it was removed or is now needed.

**Expected behaviour**:
I would run my kubectl apply and helm upgrade to upgrade everything and it would work as it did before but with the updated features.

**Steps to reproduce the bug**:
Install Vault Operator Helm Chart v0.2.6, with Bank-Vaults v0.4.6 and Vault v1.0.2 on kubernetes using kubernetes auth.  Then upgrade to Bank-Vaults v0.5.1, Vault v1.2.2 and Vault Operator Helm Chart v0.4.1.  Try to login via kubernetes once the upgrade is complete.

**Additional context**:

**Environment details**:
- Kubernetes version (e.g. v1.10.2): v1.8.3
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): AWS
- bank-vaults version (e.g. 0.4.17): 0.4.6 -> 0.5.1
- Install method (e.g. helm or static manifests): Helm
- Logs from the misbehaving component (and any other relevant logs): 
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:

```
apiVersion: "vault.banzaicloud.com/v1alpha1"
kind: "Vault"
metadata:
  name: vault
  namespace: {namespace}
spec:
  size: 2
  image: vault:1.2.2
  bankVaultsImage: banzaicloud/bank-vaults:0.5.1
  serviceType: ClusterIP

  # Insted of credentialsConfig one can use IAM instance profiles, or kube2iam for example:
  #annotations:
  #  iam.amazonaws.com/role: "role-arn..."

  # Describe where you would like to store the Vault unseal keys and root token
  # in S3 encrypted with KMS.
  unsealConfig:
    aws:
      kmsKeyId: ""
      kmsRegion: ""
      s3Bucket: ""
      s3Prefix: ""
      s3Region: ""

  # Specify the ServiceAccount where the Vault Pod and the Bank-Vaults configurer/unsealer is running
  serviceAccount: vault-operator

  # A YAML representation of a final vault config file, this config represents
  # a backend config in AWS.
  # See https://www.vaultproject.io/docs/configuration/ for more information.
  config:
    storage:
      dynamodb:
        ha_enabled: "true"
        region: {region}
        table: {name}
    listener:
      tcp:
        address: "0.0.0.0:8200"
        tls_disable: true

        # For TLS, we need to solve the issue of rotating keys.
        # Vault certs expire after 1 year and are also self signed.
        # So if we import and trust the self-signed certs, we'd have to
        # remember to update the trust store and deploy (e.g.
        # update in the various containers and deploy the update).
        #tls_cert_file: /vault/tls/server.crt
        #tls_key_file: /vault/tls/server.key
    api_addr: {address}
    telemetry:
      statsd_address: {gateway address}
    ui: true

  # See: https://github.com/banzaicloud/bank-vaults#example-external-vault-configuration for more details.
  externalConfig:
    policies:
      - name: read_secrets
        rules: path "secret/data/*" {
            capabilities = ["read"]
          }
      - name: write_secrets
        # TODO: support `kv2` permissions for deleting, undeleting, etc?
        rules: path "secret/data/*" {
            capabilities = ["create", "update", "read", "delete"]
          }
    auth:
      - type: kubernetes
        roles:
          # Allow every pod, running as the `default` service account, in all namespaces, to read secrets
          - name: {name}
            bound_service_account_names: "*"
            bound_service_account_namespaces: “{spaces}”
            policies: {policy}
            ttl: 1h
      - type: aws
        config:
          iam_server_id_header_value: {value}
        roles:
          - name: {name}
            bound_iam_principal_arn: {arn}
            ttl: 1h
            policies: {policy}
          
  # If you are not using IAM instance profiles to authenticate against AWS you can pass in an
  # AWS credentials refering to a Secret with the following configuration block. You have to
  # create the "aws-cred" secret manually, it has to have a key called "credentials" holding
  # the configuration in the following format:
  # https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#the-shared-credentials-file
  credentialsConfig:
    env: AWS_SHARED_CREDENTIALS_FILE
    path: /etc/aws/credentials
    secretName: {name}
```

/kind bug

**Comment by pbalogh-sa** [comment]
@jordan-next 

> We believe our issue has to do with missing cluster roles definitions. Compared to our other environments one big thing missing is vault-operator-auth-delegator which was removed in this commit. Is this something that is no longer needed or needs to be created in the newer vault CRs?

The auth-delegator cluster role isn't needed to vault-operator.
The `system:auth-delegator` cluster role is necessary only for your vault pods, so you have to bind the `serviceAccount` you defined in CR to this `ClusterRole`. In your case this `seviceAccount` is vault-operator:

>  # Specify the ServiceAccount where the Vault Pod and the Bank-Vaults configurer/unsealer is running
  serviceAccount: vault-operator

The service account which used by vault pods is up to you.

**Comment by jordan-next** [comment]
Thank you, looks like something changed in our upgrade.  I added it in our steps to make sure thatt a new service account is created when we deploy if needed.
